openest/generate/smart_curve.py

```diff
@@ -258,13 +260,14 @@ def __call__(self, ds):
             return result
         except Exception as ex:
             if self.allow_raising:
-                return curve.CubicSplineCurve(self.knots, self.coeffs)(ds._variables[self.variables[0]]._data)
+                import traceback;print("".join(traceback.format_exception(ex.__class__, ex, ex.__traceback__)))  # CATBELL
```

Comment:
Why do we want this here?

Reply:
We don't want this here. Unless you can tell me exactly what exceptions we should catch, we *need* this here until we get this sorted out.

This is here because we need to identify the actual exceptions we want to catch so we don't allow this to trigger on *every* exception -- which is a horrible horrible bug that becomes very difficult to debug. Seriously. Don't catch every exception and continue executing without parsing or logging the error.

This logs exactly what exception was fired and from  where in the stack.

Once we've narrowed the exceptions we're trying to catch - then we can remove printed crap.

Reply:
Ah, we only want to catch `KeyError` exceptions coming from `ds._variables[x]`.

Reply:
@jrising Great! I can swap that in then!

Reply:
Resolved with bfec054.